src/main/java/uk/gov/hmcts/cmc/claimstore/services/ccd/CallbackService.java

```diff
@@ -0,0 +1,44 @@
+package uk.gov.hmcts.cmc.claimstore.services.ccd;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.cmc.ccd.domain.CaseEvent;
+import uk.gov.hmcts.cmc.claimstore.exceptions.CallbackException;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.CallbackType;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.GenerateOrderCallbackService;
+import uk.gov.hmcts.cmc.claimstore.services.ccd.callbacks.MoreTimeRequestedCallbackService;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackRequest;
+import uk.gov.hmcts.reform.ccd.client.model.CallbackResponse;
+
+import static java.lang.String.format;
+
+@Component
+public class CallbackService {
+
+    private final GenerateOrderCallbackService generateOrderCallbackService;
+    private final MoreTimeRequestedCallbackService moreTimeRequestedCallbackService;
+
+    @Autowired
+    public CallbackService(
+        MoreTimeRequestedCallbackService moreTimeRequestedCallbackService,
+        GenerateOrderCallbackService generateOrderCallbackService) {
+        this.moreTimeRequestedCallbackService = moreTimeRequestedCallbackService;
+        this.generateOrderCallbackService = generateOrderCallbackService;
+    }
+
+    public CallbackResponse dispatch(
+        String authorisation,
+        CallbackType callbackType,
+        CallbackRequest callbackRequest) {
+        CaseEvent caseEvent = CaseEvent.fromValue(callbackRequest.getEventId());
+        switch (caseEvent) {
```

Comment:
How about creating a `CallbackFactory` instead?

Reply:
Factory or some generic way of handling event, as we have got more callbacks event coming down the line. However that can go in separate PR.

Reply:
I originally had it like that and refactored because it looked a bit wasteful, but I agree, let's do it :) we are going to have more callbacks coming in anyway